app/controllers/api/products_controller.rb

```diff
@@ -121,16 +121,17 @@ def product_query_includes
     def paged_products_for_producers(producers)
       Spree::Product.scoped.
         merge(product_scope).
+        includes(variants: [:product, :default_price, :stock_items]).
```

Comment:
so it's this what saves those many queries?

Reply:
There were some huge N+1s here, yes.